src/tabs/tab-header-bar.tsx

```diff
@@ -122,26 +124,23 @@ export function TabHeaderBar({
     [styles['pagination-button-right-scrollable']]: rightOverflow,
   });
 
-  const paginationButtonAttributes = {
-    'aria-hidden': true,
-    tabIndex: -1,
-  };
-
   return (
     //converted span to div as list should not be a child of span for HTML validation
     <div className={classes} ref={containerRef}>
-      {horizontalOverflow && (
-        <span ref={leftOverflowButton} className={leftButtonClasses}>
-          <InternalButton
-            formAction="none"
-            variant="icon"
-            iconName="angle-left"
-            __nativeAttributes={paginationButtonAttributes}
-            disabled={!leftOverflow}
-            onClick={() => onPaginationClick(headerBarRef, -1)}
-          />
-        </span>
-      )}
+      {
+        /* istanbul ignore next: covered by integration tests */ horizontalOverflow && (
```

Comment:
Is there really no way to test this (at least partially) in a unit test?

Reply:
Mocking seems to work, however my two main concerns are: 1) relying on implementation details (the functions I'm mocking), and 2) relying too much on Jest features.

Also tried just wrapping the component in a narrow parent, which would have been much easier and simpler but in the unit tests environment unfortunately it didn't work.

Reply:
Added unit tests with mocking in latest commits, and removed equivalent integration tests